.azure/pipelines/jobs/demos.yml

```diff
@@ -0,0 +1,23 @@
+jobs:
+  - job: demos
+    pool:
+      vmImage: ubuntu-20.04
+    container: dotnet-sdk-gnuplot5
+    variables:
+      SKIP_PLOT_DISPLAY: true
+    steps:
+      - bash: |
+          cd demos
+          dotnet restore
+          dotnet build --no-restore
+        displayName: build
+
+      - bash: |
+          cd demos
+          chmod +x run.sh
+          ./run.sh
+        displayName: run demos
+
+      - publish: demos/*/**/*.png
```

Comment:
The [RTFM](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/utility/publish-pipeline-artifact?view=azure-devops#arguments) says (emphasis mine):

> Path to the folder or file you want to publish. **Wildcards are not supported**. The path must be a fully-qualified path or a valid path relative to the root directory of your repository.